### ISSUE TITLE
Ensure mutation happens

### ISSUE THREAD

**Issue by mficek** [rhiever/tpot#364]
I'm checking how function `varOr` works and wonder, if mutation (or crossover) with no effect is OK to have. 

## Context of the issue

In Function `varOr`, there are three branches, that get selected by probability `cxpb` and `mutpb`. But the code allows for no mutation (crossover) to happen:
```
            ind, = toolbox.mutate(ind)
            if ind_str != str(ind): # check if mutation happend
                del ind.fitness.values
```
When no mutation happens? Obviously for example if mutation is "shrink" and there is nothing to shrink. But then: wouldn't it be better to have something like this:
``` 
    while ind_str == str(ind):
        ind, = toolbox.mutate(ind)
    del ind.fitness values
```
i.e., ensure, that a mutation happens always?
Now, if mutation does not happen, it means it is basically a reproduction and it means, that our probability `mutpb` does not hold anymore, it's lower. 

I can see one drawback of the solution I'm proposing: imagine we have a classifier with no parameters, no other classifiers and (hypothetically) no selector and preprocessor in `config_classifier.py`. Then this cycle would never end. 

What do you think?

**Comment by rhiever** [comment]
@weixuanfu2016 can confirm, but I believe we've put in checks to make sure that a mutation or crossover actually changes the individual. It retries something like 100x (instead of infinitely) to avoid the infinite loop situation you describe.

**Comment by weixuanfu2016** [comment]
It was a [infinitely loop](https://github.com/weixuanfu2016/tpot/blob/point_mut/tpot/base.py#L713-714) in one of a old PR. But I removed this one in new dev batch due to this issue. I will submit another PR later this afternoon to solve this issue. Thank @mficek for catching it.

**Comment by weixuanfu2016** [comment]
PR #365 is made for this issue. It will rerun mutation and crossover 50 times at most to force them to generate a different child from its patients. But I still need keep `if ind_str != str(ind)` because:

1. as you mentioned, a classifier with no parameters cannot mutate.
2. For two parents which both are single-operator "pipeline", crossover does not work for generate a new individual.

**Comment by mficek** [comment]
Resolved with PR #365. Thanks! Closing.
